drivers/flash/flash_stm32f4x.c

```diff
@@ -292,8 +292,7 @@ int flash_stm32_update_rdp(const struct device *dev, bool enable,
 	switch (current_level) {
 	case FLASH_STM32_RDP2:
 		if (!enable || !permanent) {
-			__ASSERT(false, "RDP level 2 is permanent and can't be "
-					"changed!");
+			LOG_ERR("RDP level 2 is permanent and can't be changed!");
```

Comment:
I wonder if we really need `LOG_ERR` here. These messages consumes flash, but in well written application they will be never printed. How about changing it `LOG_WRN`? What do you think?

Reply:
Okay.